tools/smoke/internal/smoke.go

```diff
@@ -37,6 +40,7 @@ func (c *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&c.Kubeconfig, "kubeconfig", DefaultConfig.Kubeconfig, "absolute path to the kubeconfig file")
 	f.StringVar(&c.PodPrefix, "pod-prefix", DefaultConfig.PodPrefix, "prefix for grafana agent pods")
 	f.BoolVar(&c.FakeRemoteWrite, "fake-remote-write", DefaultConfig.FakeRemoteWrite, "remote write endpoint for series which are discarded, useful for testing and not storing metrics")
+	f.BoolVar(&c.SimulateErrors, "simulate-errors", DefaultConfig.SimulateErrors, "remote write endpoint will return a 500 error response 1% of the time.")
```

Comment:
Nice, should the percentage be configurable though? Maybe something like `--simulate-error-percentage=0.01` (for 1%, expecting a value between 0 and 1)?

Reply:
Yeah, I could add this. The way I've implemented the 1% is basically to just return an error every 100th request. I think it might make sense to use the error percentage as a weighted probability so if it's set to 0.05, then there's a 5% chance each request will return an error. What do you think?